core/lib.rs

```diff
@@ -192,8 +192,9 @@ impl Database {
             unsafe { &*wal.get() }.max_frame.load(Ordering::SeqCst) > 0
         });
 
-        let is_empty = if db_size == 0 && !wal_has_frames {
-            DB_STATE_UNITIALIZED
+        // No pages in DB file or WAL -> empty database
+        let db_state = if db_size == 0 && !wal_has_frames {
```

Comment:
I think since we init DB on first transaction these days, `db_size == 0` should be enough.